packages/webpack/src/config/client.js

```diff
@@ -30,13 +30,17 @@ export default class WebpackClientConfig extends WebpackBaseConfig {
   }
 
   env () {
-    return Object.assign(super.env(), {
-      'process.env.VUE_ENV': JSON.stringify('client'),
-      'process.browser': true,
-      'process.client': true,
-      'process.server': false,
-      'process.modern': false
-    })
+    return Object.assign(
+      super.env(),
+      {
+        'process.env.VUE_ENV': JSON.stringify('client'),
+        'process.browser': true,
+        'process.client': true,
+        'process.server': false,
+        'process.modern': false
+      },
+      ...(this.buildContext.buildOptions.aggressiveClientServerCodeRemoval && { 'typeof window': JSON.stringify('object') })
```

Comment:
I think this can be moved to `base.js::env`:

```js
if(this.buildContext.buildOptions...) {
	env['typeof window'] = this.isServer
}
```

Reply:
Couldn't `	env['typeof process'] = this.isServer` possibly remove valid code if it runs before webpack define plugin?

Reply:
I didn’t get you, can you give me example?

Reply:
Nvm, I thought twice about the use case I had in mind and it'll work nevertheless :relaxed: